src/main/java/org/kaazing/netx/ws/internal/WsURLConnectionImpl.java

```diff
@@ -108,6 +109,9 @@
     private final WebSocketInputStateMachine inputStateMachine;
     private final WebSocketOutputStateMachine outputStateMachine;
     private final WebSocketExtensionFactory extensionFactory;
+    private final OptimisticReentrantLock readLock;
+    private final OptimisticReentrantLock stateLock;
+    private final OptimisticReentrantLock writeLock;
 
     private String negotiatedProtocol;
     private WsInputStream inputStream;
```

Comment:
Do these non-final fields need to be `volatile` to make sure different threads see the initialized values even if they are racing?